pypeit/reduce.py

```diff
@@ -681,26 +785,37 @@ def local_skysub_extract(self, waveimg, global_sky, sobjs,
 
         # Could actually create a model anyway here, but probably
         # overkill since nothing is extracted
-
         self.sobjs = sobjs.copy()  # WHY DO WE CREATE A COPY HERE?
         # Loop on slits
         for slit in gdslits:
             msgs.info("Local sky subtraction and extraction for slit: {:d}".format(slit))
             thisobj = (self.sobjs.SLITID == slit) # indices of objects for this slit
+            sobj = self.sobjs[thisobj]
             if np.any(thisobj):
                 thismask = (self.slitmask == slit) # pixels for this slit
                 # True  = Good, False = Bad for inmask
                 inmask = (self.sciImg.mask == 0) & thismask
-                # Local sky subtraction and extraction
-                self.skymodel[thismask], self.objmodel[thismask], self.ivarmodel[thismask], \
-                    self.extractmask[thismask] = skysub.local_skysub_extract(
-                    self.sciImg.image, self.sciImg.ivar, self.tilts, self.waveimg, self.global_sky, self.sciImg.rn2img,
-                    thismask, self.tslits_dict['slit_left'][:,slit], self.tslits_dict['slit_righ'][:, slit],
-                    self.sobjs[thisobj], spat_pix=spat_pix, model_full_slit=self.redux_par['model_full_slit'],
-                    box_rad=self.redux_par['boxcar_radius']/self.spectrograph.detector[self.det-1]['platescale'],
-                    sigrej=self.redux_par['sky_sigrej'],
-                    model_noise=model_noise, std=std, bsp=self.redux_par['bspline_spacing'],
-                    sn_gauss=self.redux_par['sn_gauss'], inmask=inmask, show_profile=show_profile)
+                if self.extraction_par['boxcar_only']:
```

Comment:
I would choose a different name for this parameter. Like "no_local_skysub" or "global_skysub_only". The point is not really that we only do boxcar, it is that we skip the local sky sub entirely.

Reply:
Well, what if we want to do Optimal but not local skysub?

It seems that ought to be an option, although I guess that
would be hard given the way the code is now.

How about:   `skip_optimal` or `skip_optimal_and_local`?

Reply:
I like skip_optimal

Reply:
Done.